src/api/codewriter.rs

```diff
@@ -0,0 +1,153 @@
+use byteorder::{BigEndian, WriteBytesExt};
+use std::fs::File;
+use std::io::prelude::*;
+use std::io::Error;
+use std::result::Result;
+
+use crate::objects::codeholder::CodeHolder;
+use crate::objects::instruction::Instruction;
+use crate::objects::register::{Register, RegisterLocation, RegisterReference};
+
+fn write_register(buf: &mut Vec<u8>, r: &Register) -> Result<(), Error> {
+    buf.write_u32::<BigEndian>(r.0)?;
+    buf.push(match r.1 {
+        RegisterLocation::ConstantPool => 0x01,
```

Comment:
Since magic numbers are used across the reader and writer, I think it would be best to have a "parseing_constants" file that contains the constants needed across both the reader and writer